superbench/benchmarks/model_benchmarks/pytorch_lstm.py

```diff
@@ -0,0 +1,196 @@
+
+        self._parser.add_argument(
+            '--num_classes', type=int, default=100, required=False, help='The number of objects for classification.'
+        )
+        self._parser.add_argument(
+            '--input_size', type=int, default=256, required=False, help='The number of expected features in the input.'
+        )
+        self._parser.add_argument(
+            '--hidden_size', type=int, default=1024, required=False, help='The number of features in the hidden state.'
+        )
+        self._parser.add_argument(
+            '--num_layers', type=int, default=8, required=False, help='The number of recurrent layers.'
+        )
+
+        self._parser.add_argument('--bidirectional', action='store_true', default=False, help='Bidirectional LSTM.')
+        self._parser.add_argument('--seq_len', type=int, default=512, required=False, help='Sequence length.')
+
+    def _generate_dataset(self):
+        """Generate dataset for benchmarking according to shape info.
+
+        Return:
+            True if dataset is created successfully.
+        """
+        self._dataset = TorchRandomDataset(
+            [self._args.sample_count, self._args.seq_len, self._args.input_size], self._world_size, dtype=torch.float
+        )
+        if len(self._dataset) == 0:
+            logger.error('Generate random dataset failed - model: {}'.format(self._name))
+            return False
+
+        return True
+
+    def _create_model(self, precision):
+        """Construct the model for benchmarking.
+
+        Args:
+            precision (Precision): precision of model and input data, such as float32, float16.
+        """
+        try:
+            self._model = LSTMBenchmarkModel(
+                self._args.input_size, self._args.hidden_size, self._args.num_layers, self._args.bidirectional,
+                self._args.num_classes
+            )
+            self._model = self._model.to(dtype=getattr(torch, precision.value))
+            if self._gpu_available:
+                self._model = self._model.cuda()
+        except BaseException as e:
+            logger.error(
+                'Create model with specified precision failed - model: {}, precision: {}, message: {}.'.format(
+                    self._name, precision, str(e)
+                )
+            )
+            return False
+
+        self._target = torch.LongTensor(self._args.batch_size).random_(self._args.num_classes)
+        if self._gpu_available:
+            self._target = self._target.cuda()
+
+        return True
+
+    def _train_step(self, precision):
+        """Define the training process.
+
+        Args:
+            precision (Precision): precision of model and input data, such as float32, float16.
+
+        Return:
+            The step-time list of every training step.
+        """
+        duration = []
+        curr_step = 0
+        while True:
+            for idx, sample in enumerate(self._dataloader):
+                start = time.time()
```

Comment:
could we put this into `PytorchBase` if this function is nearly the same for all models?

Reply:
Let's consider extracting the common logic in a separate PR for all models.